src/linter.js

```diff
@@ -154,104 +152,43 @@ export default class Linter {
-          },
-          message: {
-            headingTransform: () => {
-              return i18n._('Message');
-            },
-            maxWidth: (maxColumns - 35) * 0.25,
-          },
-          description: {
-            headingTransform: () => {
-              return i18n._('Description');
-            },
-            maxWidth: (maxColumns - 35) * 0.5,
-          },
-          file: {
-            headingTransform: () => {
-              return i18n._('File');
-            },
-            maxWidth: (maxColumns - 35) * 0.25,
-          },
-          line: {
-            headingTransform: () => {
-              return i18n._('Line');
-            },
-            maxWidth: 6,
-          },
-          column: {
-            headingTransform: () => {
-              return i18n._('Column');
-            },
-            maxWidth: 6,
-          },
-        };
-
-        const outputColumns = [
-          'code',
-          'message',
-          'description',
-          'file',
-          'line',
-          'column',
-        ];
-
-        // If the terminal is this small we cave and don't size things
-        // contextually anymore.
-        if (maxColumns < 60) {
-          delete outputColumns[outputColumns.indexOf('column')];
-          delete outputConfig.column;
-          delete outputColumns[outputColumns.indexOf('description')];
-          delete outputConfig.description;
-          delete outputColumns[outputColumns.indexOf('line')];
-          delete outputConfig.line;
-
-          outputConfig.message.maxWidth = 15;
-          outputConfig.file.maxWidth = 15;
-        } else if (maxColumns < 78) {
-          delete outputColumns[outputColumns.indexOf('description')];
-          delete outputConfig.description;
-
-          outputConfig.message.maxWidth = (maxColumns - 47) * 0.5;
-          outputConfig.file.maxWidth = (maxColumns - 35) * 0.5;
-        }
-
-        out.push(`${messageType.toUpperCase()}:`);
+        out.push(this.colorize(type)(`${messageType.toUpperCase()}:`));
         out.push('');
-        out.push(
-          columnify(this.output[messageType], {
-            maxWidth: 35,
-            columns: outputColumns,
-            columnSplitter: '   ',
-            config: outputConfig,
-          })
-        );
+
+        const groupedMessages = {};
+        this.output[messageType].forEach((message) => {
+          if (!groupedMessages[message.code]) {
+            groupedMessages[message.code] = [];
+          }
+          groupedMessages[message.code].push(message);
+        });
+
+        Object.entries(groupedMessages).forEach(([code, list]) => {
+          out.push(this.colorize(type)(`  ${code}`));
+          out.push(`  ${list[0].message}`);
+          list.forEach((message) => {
+            const location = message.file
+              ? `${message.file}${message.line ? `:${message.line}` : ''}${
+                  message.column ? `:${message.column}` : ''
+                }`
+              : 'N/A';
```

Comment:
Could you split this string construction in separate variables, so that there are no three ternary expressions embedded in one?

For example something like the following (untested, but just to show how it could be unrolled into something that is a bit more readable):

```suggestion
            let location = message.file || 'N/A';
            if (message.file && message.line) {
              location += `:${message.line}`;
              if (message.column) {
                location += `:${message.column}`; 
              }
            }
```